src/test/ProposalPayloadE2E.t.sol

```diff
@@ -3,37 +3,335 @@ pragma solidity ^0.8.15;
+    IReserveInterestRateStrategyV2 strategyPolygon = IReserveInterestRateStrategyV2(NEW_INTEREST_RATE_STRATEGY_POLYGON);
 
     function setUp() public {
         // To fork at a specific block: vm.createSelectFork(vm.rpcUrl("mainnet"), BLOCK_NUMBER);
-        vm.createSelectFork(vm.rpcUrl("mainnet"));
+        mainnetFork = vm.createFork(vm.rpcUrl("mainnet"), 16435723);
+        polygonFork = vm.createFork(vm.rpcUrl("polygon"), 38278604);
 
-        // Deploy Payload
-        ProposalPayload proposalPayload = new ProposalPayload();
+        // Deploy Payloads
+        vm.selectFork(polygonFork);
+        proposalPayloadPolygon = new ProposalPayloadPolygon();
+
+        vm.selectFork(mainnetFork);
+        proposalPayload = new ProposalPayload();
 
         // Create Proposal
         vm.prank(GovHelpers.AAVE_WHALE);
-        proposalId = DeployMainnetProposal._deployMainnetProposal(
+        proposalId = DeployProposals._deployMainnetProposal(
             address(proposalPayload),
+            address(proposalPayloadPolygon),
             0x344d3181f08b3186228b93bac0005a3a961238164b8b06cbb5f0428a9180b8a7 // TODO: Replace with actual IPFS Hash
         );
     }
 
+    // Utility to transform memory to calldata so array range access is available
+    function _cutBytes(bytes calldata input) public pure returns (bytes calldata) {
+        return input[64:];
+    }
+
     function testExecute() public {
-        // Pre-execution assertations
+        _modifyStrategy();
+
+        // Post-execution assertations
+        vm.selectFork(polygonFork);
+
+        ReserveConfig[] memory allConfigsAfterV3Polygon = _getReservesConfigs(AaveV3Polygon.POOL);
+
+        ReserveConfig memory expectedConfigPolygonV3 = ReserveConfig({
+            symbol: "BAL",
+            underlying: BAL_POLYGON,
+            aToken: address(0), // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            variableDebtToken: address(0), // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            stableDebtToken: address(0), // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            decimals: 18,
+            ltv: 2000,
+            liquidationThreshold: 4500,
+            liquidationBonus: 11000,
+            liquidationProtocolFee: 1000,
+            reserveFactor: 2000,
+            usageAsCollateralEnabled: true,
+            borrowingEnabled: true,
+            interestRateStrategy: _findReserveConfigBySymbol(allConfigsAfterV3Polygon, "BAL").interestRateStrategy,
+            stableBorrowRateEnabled: false,
+            isActive: true,
+            isFrozen: false,
+            isSiloed: false,
+            isBorrowableInIsolation: false,
+            isFlashloanable: false,
+            supplyCap: 284_600,
+            borrowCap: 256_140,
```

Comment:
Besides the current `_validateReserveConfig` check, Is there an Aave Helper that you can use to check that only the Borrow Cap is changing (similar to the `_validateInterestRateStrategy` check you have below)? 

I remember seeing some helper a while back that allows you to specify a specific property that you can check. That will make it a lot more explicit and ensure that no other properties are being changed. Currently I have no way to ensure that than to trust your `expectedConfig`

Reply:
Hey @0xrajath, unfortunately there isn't, but I made it a variable on the payload and I'm checking with that value now.